### PR TITLE
V0.6fix

### PR DESCRIPTION
**Pull request by pkofod**
<s>edit: how annoying, but of course rebase --onto rewrites the history with new ids... i'll let tests run</s> commit mess sorted out

### CODE REVIEW

File: src/cg.jl [hunk 1]
```diff
@@ -93,9 +93,9 @@ function initial_state{T}(method::ConjugateGradient, options, d, initial_x::Arra
     if !isfinite(f_x)
         error("Must have finite starting value")
     end
-    if !all(isfinite(g))
+    if !all(isfinite.(g))
```
Comment (andreasnoack) [inline src/cg.jl]: I think that `!all(isfinite, g)` will save you a temporary here.

File: test/types.jl [hunk 2]
```diff
@@ -14,7 +14,7 @@ let
 
         io = IOBuffer()
         show(io, res)
-        s = takebuf_string(io)
+        s = String(take!(io))
```
Comment (andreasnoack) [inline test/types.jl]: I think the missing
```jl
using Compat
import Compat.String
```
is causing the test failures.

**Comment by andreasnoack** [comment]
I think we opened our PRs almost simultaneously. I'll clone mine.

**Comment by pkofod** [comment]
> I think we opened our PRs almost simultaneously. I'll clone mine.

Didn't realize! I'll take a look and add some of the stuff you had in there.

**Comment by pkofod** [comment]
@andreasnoack thanks for the changes in the other PR. I added the changes I hadn't included, but I forgot to `--author` you, I hope it's OK.

**Comment by andreasnoack** [comment]
> but I forgot to `--author you`, I hope it's OK.

Of course. No problem.

File: test/levenberg_marquardt.jl [hunk 3]
```diff
@@ -50,12 +50,12 @@ let
...
         srand(12345)
 
-        model(x, p) = p[1]*exp(-x.*p[2])
+        model(x, p) = p[1]*exp.(-x.*p[2])
 
         xdata = linspace(0,10,20)
         ydata = model(xdata, [1.0 2.0]) + 0.01*randn(length(xdata))
 
-        f_lsq = p -> p[1]*exp(-xdata.*p[2])-ydata
+        f_lsq = p -> p[1]*exp.(-xdata.*p[2])-ydata
```
Comment (andreasnoack) [inline test/levenberg_marquardt.jl]: This file needs a `using Compat` to work on 0.4.

**Pull request merged**